stl/inc/__msvc_print.hpp

```diff
@@ -8,6 +8,11 @@
 #include <yvals_core.h>
 #if _STL_COMPILER_PREPROCESSOR
 
+#ifndef __cpp_lib_concepts
+#error The contents of <format> are available only with C++20 or later, and the contents of <print> \
+are available only with C++23 or later. (Also, you should not include this internal header.)

```

Comment:
I think these duplicate error messages from internal headers only confuse users. `<format>` presumably tells me when I can't include it, so I don't need `__msvc_meow.hpp` to tell me repeatedly. Why can't we consistently change the internal headers to just "You should not include this internal header."?

Reply:
There are no duplicate error messages - can you take another look?

The Standard headers emit `#pragma message` warnings and expand to nothing, so they bypass any errors from internal headers. Here's what it looks like:

```
D:\GitHub\STL\out\x64>type meow1.cpp
#include <format>

D:\GitHub\STL\out\x64>cl /EHsc /nologo /W4 /std:c++17 /c meow1.cpp
meow1.cpp
D:\GitHub\STL\out\x64\out\inc\format(42): warning STL4038: The contents of <format> are available only with C++20 or later.

D:\GitHub\STL\out\x64>type meow2.cpp
#include <print>

D:\GitHub\STL\out\x64>cl /EHsc /nologo /W4 /std:c++17 /c meow2.cpp
meow2.cpp
D:\GitHub\STL\out\x64\out\inc\print(11): warning STL4038: The contents of <print> are available only with C++23 or later.

D:\GitHub\STL\out\x64>type meow3.cpp
#include <__msvc_print.hpp>

D:\GitHub\STL\out\x64>cl /EHsc /nologo /W4 /std:c++17 /c meow3.cpp
meow3.cpp
D:\GitHub\STL\out\x64\out\inc\__msvc_print.hpp(13): fatal error C1189: #error:  The contents of <format> are available only with C++20 or later, and the contents of <print> are available only with C++23 or later. (Also, you should not include this internal header.)
```

https://github.com/microsoft/STL/blob/adea8d5ae280cafb91ae69b8dfaecd1c37a847d9/stl/inc/print#L6-L14

Reply:
I stand corrected, and it now makes even _less_ sense to me that e.g. `__msvc_print.hpp` mentions `<format>`. Why is this message informing me of the requirements of headers I didn't even include?

Reply:
I've added a comment to the guard, because otherwise it looks strange that we're checking for 20-concepts when `<print>` needs 23-concepts, and otherwise rephrased the message to follow the others (dropping "or later", as the others do, which makes it fit on one line).